jdbc/src/main/java/org/openconnectors/JdbcSourceConnector.java

```
@@ -0,0 +1,192 @@
+
+        Set<String> whiteListTables = getWhiteListTables();
+        Set<String> blackListTables = getBlackListTables();
+        if (!whiteListTables.isEmpty() && !blackListTables.isEmpty()) {
+            String errorMsg = "Only white list or black list of tables must be specified, rather than both of them.";
+            logger.error(errorMsg);
+            throw new RuntimeException(errorMsg);
+        }
+        List<String> tables = new ArrayList<>();
+        String schemaPattern = this.config.getString(JdbcConfigKeys.SCHEMA_PATTERN);
+        // TODO: handle types[] parameter
+        ResultSet tablesResultSet = connection.getMetaData().getTables(null, schemaPattern, "%", null);
+        while (tablesResultSet.next()) {
+            String tableName = tablesResultSet.getString(3).toLowerCase();
+            tables.add(tableName);
+        }
+
+        List<String> filteredTables = new ArrayList<>();
+        if (!whiteListTables.isEmpty()) {
+            filteredTables.addAll(tables.stream()
+                    .filter(whiteListTables::contains)
+                    .collect(Collectors.toList())
+            );
+        }
+
+        if (!blackListTables.isEmpty()) {
+            filteredTables.addAll(tables.stream()
+                    .filter(table -> !blackListTables.contains(table))
+                    .collect(Collectors.toList())
+            );
+        }
+        String mode = this.config.getString(JdbcConfigKeys.MODE);
+        List<TableQuerier> tableQueriers = new ArrayList<>();
+        for (String tableName : filteredTables) {
+            switch (mode) {
+                case JdbcConfigKeys.BULK_MODE:
+                    tableQueriers.add(
+                            new BulkJdbcQuerier(getConnectionProvider(connectionConfig), schemaPattern, tableName)
+                    );
+                    break;
+                case JdbcConfigKeys.AUTOINCREMENTING_MODE:
+                    //TODO: implement
```

better to throw an error saying not yet implemented?